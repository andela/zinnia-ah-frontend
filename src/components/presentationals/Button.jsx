import React from 'react';
import PropTypes from 'prop-types';

import './Button.scss';

const Button = props => {
  return (
    <button className={`btn ${props.button.class}`} type={props.button.class}>
      {props.button.value}
    </button>
  );
};

Button.propTypes = {
  button: PropTypes.shape({
    class: PropTypes.string,
    content: PropTypes.string,
    value: PropTypes.string,
  }),
};

export default Button;
