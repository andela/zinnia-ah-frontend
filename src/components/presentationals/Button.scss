@import url("https://fonts.googleapis.com/css?family=Lato:400,700|Lora:400,700|Sacramento");
@import "../../assets/variables";

.btn {
  background-color: $secondary-grey;
  border-radius: 5px;
  color: $primary-black;
  cursor: pointer;
  font-family: "Lato", sans-serif;
  font-size: 14px;
  height: 50px;
  letter-spacing: 0.1em;
  text-align: center;
  width: 100%;
}

.btn-dark {
  background-color: $primary-black;
  border: 1px solid $primary-black;
  color: $primary-white;
}

.btn-white {
  background-color: $primary-white;
  border: 1px solid $primary-black;
  color: $primary-black;
}
