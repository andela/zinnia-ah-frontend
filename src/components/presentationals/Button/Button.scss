@import url('https://fonts.googleapis.com/css?family=Lato:400,700|Lora:400,700|Sacramento');
@import '../../../assets/variables';

.btn {
  background-color: $secondary-grey;
  border-radius: .3125rem;
  color: $primary-black;
  cursor: pointer;
  font-family: 'Lato', sans-serif;
  font-size: .875rem;
  height: 3.125rem;
  letter-spacing: .1em;
  outline: none;
  text-align: center;
  width: 100%;
}

.btn-dark {
  background-color: $primary-black;
  border: .0625rem solid $primary-black;
  color: $primary-white;
}

.btn-white {
  background-color: $primary-white;
  border: .0625rem solid $primary-black;
  color: $primary-black;
}
