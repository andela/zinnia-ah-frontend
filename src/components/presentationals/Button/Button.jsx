import React from 'react';
import PropTypes from 'prop-types';

import './Button.scss';

const Button = ({ className, type, onClick, value, children }) => {
  return (
    <button className={`btn ${className}`} type={type} onClick={onClick}>
      {value ? value : children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.any,
  className: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  type: PropTypes.string,
  value: PropTypes.string.isRequired,
};

export default Button;
