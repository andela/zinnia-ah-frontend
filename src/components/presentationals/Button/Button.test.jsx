import React from 'react';
import { mount } from 'enzyme';

import Button from './Button';

describe('Button', () => {
  it('should render correctly', () => {
    const component = mount(
      <Button className="btn" type="button" value="login" />,
    );
    expect(component).toMatchSnapshot();
  });
});
