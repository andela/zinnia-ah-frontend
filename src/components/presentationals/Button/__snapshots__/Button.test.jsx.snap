// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<Button
  className="btn"
  type="button"
  value="login"
>
  <button
    className="btn btn"
    type="button"
  >
    login
  </button>
</Button>
`;
